tests/tools/test_dist_part.py

```diff
@@ -0,0 +1,232 @@
+            {
+            'feat': paper_feat_path,
+            'label': paper_label_path,
+            'year': paper_year_path
+            }
+        },
+        {
+            'cites': {'count': cite_count_path},
+            'writes': {'year': write_year_path},
+            # you can put the same data file if they indeed share the features.
+            'rev_writes': {'year': write_year_path}
+        },
+        num_chunks=num_chunks,
+        output_path=output_dir)
+
+    # check metadata.json
+    json_file = os.path.join(output_dir, 'metadata.json')
+    assert os.path.isfile(json_file)
+    with open(json_file, 'rb') as f:
+        meta_data = json.load(f)
+    assert meta_data['graph_name'] == 'mag240m'
+    assert len(meta_data['num_nodes_per_chunk'][0]) == num_chunks
+
+    # check edge_index
+    output_edge_index_dir = os.path.join(output_dir, 'edge_index')
+    for utype, etype, vtype in data_dict.keys():
+        fname = ':'.join([utype, etype, vtype])
+        for i in range(num_chunks):
+            chunk_f_name = os.path.join(output_edge_index_dir, fname + str(i) + '.txt')
+            assert os.path.isfile(chunk_f_name)
+            with open(chunk_f_name, 'r') as f:
+                header = f.readline()
+                num1, num2 = header.rstrip().split(' ')
+                assert isinstance(int(num1), int)
+                assert isinstance(int(num2), int)
+
+    # check node_data
+    output_node_data_dir = os.path.join(output_dir, 'node_data', 'paper')
+    for feat in ['feat', 'label', 'year']:
+        for i in range(num_chunks):
+            chunk_f_name = '{}-{}.npy'.format(feat, i)
+            chunk_f_name = os.path.join(output_node_data_dir, chunk_f_name)
+            assert os.path.isfile(chunk_f_name)
+            feat_array = np.load(chunk_f_name)
+            assert feat_array.shape[0] == num_papers // num_chunks
+
+    # check edge_data
+    num_edges = {
+        'paper:cites:paper': num_cite_edges,
+        'author:writes:paper': num_write_edges,
+        'paper:rev_writes:author': num_write_edges
+    }
+    output_edge_data_dir = os.path.join(output_dir, 'edge_data')
+    for etype, feat in [
+        ['paper:cites:paper', 'count'],
+        ['author:writes:paper', 'year'],
+        ['paper:rev_writes:author', 'year']
+    ]:
+        output_edge_sub_dir = os.path.join(output_edge_data_dir, etype)
+        for i in range(num_chunks):
+            chunk_f_name = '{}-{}.npy'.format(feat, i)
+            chunk_f_name = os.path.join(output_edge_sub_dir, chunk_f_name)
+            assert os.path.isfile(chunk_f_name)
+            feat_array = np.load(chunk_f_name)
+            assert feat_array.shape[0] == num_edges[etype] // num_chunks
+
+def test_partition():
+    # Step1: graph partition
+    output_dir = '2parts'
+    os.system('python tools/partition_algorithms/random_partition.py '\
+              '--metadata chunked-data/metadata.json --output_path {} --num_partitions 2'.format(output_dir))
```

Comment:
Look like this is not following the command usage in the [doc](https://docs.dgl.ai/en/latest/guide/distributed-preprocessing.html#step-1-graph-partitioning). Could you fix it?

Reply:
We need to fix the doc instead. I'll open a separate PR for the doc fix.

Reply:
Fixed in #4433

Reply:
Changed `partition_algorithms/random_partition.py` to `partition_algo/random.py`

Reply:
Got it. `random_partition` is fine.

Reply:
Changed back. I'll update the user guide after we merge this PR.